docs/examples/CMakeLists.txt

```diff
@@ -40,5 +40,8 @@ foreach(example IN LISTS check_PROGRAMS)
   if(LIB_SELECTED STREQUAL LIB_STATIC AND WIN32)
     set_property(TARGET ${example_target} APPEND PROPERTY COMPILE_DEFINITIONS "CURL_STATICLIB")
   endif()
+  if(MSVC)
+    set_property(TARGET ${example_target} APPEND PROPERTY COMPILE_DEFINITIONS "_WINSOCK_DEPRECATED_NO_WARNINGS")  # for inet_addr()
```

Comment:
Maybe disable the warning only for the one example which emits it? As the warning is mostly about old IPv6-incompatible functions, I think it's valuable even in examples.

Reply:
I agree, it's just a single source that's affected.

Once the tests pass in #13766, I'll apply the patch to this PR.

It applies this macro to all compilers (not just MSVC) just in
case some would rely on the Microsoft SDK headers (or would
replicate this behaviour). Or do you think it's better to restrict
to `_MSC_VER`?

Reply:
Looks good to me as is now, thanks!